### PR TITLE
docs and unit tests

### PR DESCRIPTION
**Pull request by TimPurdum**
- Changed all XML references from `ArcGIS JS API` to `ArcGIS Maps SDK for JavaScript`
- bumped version to `2.3.4-beta-1`
- Refactored Blazor Test Runner to be more straightforward and simple to use.
- Added `Tests.md` that explains how to write blazor tests.

### CODE REVIEW

File: test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/GraphicsTests.razor [hunk 1]
```diff
@@ -5,49 +5,50 @@
...
+                                <Point X="0" Y="0" />
+                            </Graphic>
+                        </Graphics>
+                    </GraphicsLayer>
+                </Map>
+            </MapView>);
+
         await WaitForMapToRender();
+        Assert.AreEqual(1, graphicsLayer!.Graphics.Count);
+        graphic!.Refresh();
```
Comment (seahro) [inline test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/GraphicsTests.razor:50]: This is tied to the adjustment in MapComponent.razor.cs to adjust the Refresh()method to add `InvokeAsync`?  Just verifying.

**Review by seahro — APPROVED** [review]
Looks good

**Review by seahro — APPROVED** [review]
Looks good

File: test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/GraphicsTests.razor (hunk 1 above)
Comment (TimPurdum) [reply test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/GraphicsTests.razor:50]: Yes, exactly. This test ran on a background thread and threw an exception on this line. Thanks for checking carefully. I don't think there is any downside to calling `InvokeAsync` for intentional UI updates.

**Pull request merged**